rasa/nlu/utils/spacy_utils.py

```diff
@@ -88,21 +83,82 @@ def cache_key(
         cls, component_meta: Dict[Text, Any], model_metadata: "Metadata"
     ) -> Optional[Text]:
 
-        # Fallback, use the language name, e.g. "en",
-        # as the model name if no explicit name is defined
-        spacy_model_name = component_meta.get("model", model_metadata.language)
+        spacy_model_name = cls.check_model_fallback(
+            component_meta.get("model"), model_metadata.language, warn=True
```

Comment:
Shouldn't we always warn except in this case?

Reply:
Do we want to raise this warning when a user loads a model from disk? 

In my mind we want to raise this warning when we are training the model and this should only be raised once. The `_check_model_fallback` method is called in `SpacyNLP.create`, `SpacyNLP.load` and `SpacyNLP.cache_key` so I added the flag in order to not overflow the users retina with warnings. I can add it in all places, but the user will see the error multiple times during training.